### ISSUE TITLE
Not converging on custom dataset

### ISSUE THREAD

**Issue by fierval** [NVIDIA/retinanet-examples#98]
I have a very small dataset and need to only detect 1 class.

I don't really care about results at this point, this is just a sanity check and it's failing.

Annotations are PASCAL VOC created by labelimg tool and then converted to JSON. Verified correctness of the annotations.

Running on Titan V, Ubuntu 18.04, docker container. AP is ibarely ~ 0.01. Tried varying available hyperparameters (batch, scale, lr, backbone) - no effect. The implementation at https://github.com/fizyr/keras-retinanet works great.

**Comment by james-nvidia** [comment]
Is your class labelled as id=1 or id=0? What sizes are your images? What size are the objects you are trying to detect?

**Comment by fierval** [comment]
1. Tried with id=1 and id=0 - doesn't appear to make a difference
2. Images are 1280x720 (tried different resizing options). Also tried on images 800x500 - same results
3. Objects range from fairly large to small
The alternative implementation does fine in these settings.

I can share the dataset and annotations. It's a small dataset.

**Comment by james-nvidia** [comment]
Yes, please send me your dataset.

**Comment by fierval** [comment]
Here: https://www.dropbox.com/s/phz5jngaewxfaok/dataset.zip?dl=0

**Comment by james-nvidia** [comment]
Your dataset is so small that any computer vision model will just be memorising the data. Different implementations may do this more readily, for example if they use less regularisation, or other differences in the optimiser.
Your project looks very interesting and I'm sure would have wide commercial applicability. If you collect more data I am certain that you will be able to train an accurate model. Please reopen this issue if you have no luck with a larger dataset.

**Comment by whria78** [comment]
Hello, I am also failed to training my custom dataset with a single class. It shows mAP 0.01, too.

If I test the trained model, the range of (x,y) of the bbox exceeds the (width,height) of the test images.

I tested with pretrained models, then it's predicted bbox were within the (width,height) of the test images, but if I tested with my model, it exceeds.

I attached the trained model.
https://www.dropbox.com/s/kq4tp3jufdtljzx/5.pth?dl=0

I hope it would be helpful just to check whether the predicted bbox outbound the test images.

Thank you.
